extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionRepairAtStartTest.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.flyway.test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.flywaydb.core.Flyway;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+public class FlywayExtensionRepairAtStartTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest config = new QuarkusDevModeTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(FlywayResource.class)
+                    .addAsResource("db/migration/V1.0.0__Quarkus.sql")
+                    .addAsResource("migrate-at-start-config.properties", "application.properties"))
+            .setLogRecordPredicate(r -> true)
+            .setAllowFailedStart(true);
+
+    @Test
+    @DisplayName("Repair at start works correctly")
+    public void testRepairUsingDevMode() {
+        String version = RestAssured.get("/flyway/current-version").then().statusCode(200).extract().asString();
+        assertEquals("1.0.0", version);
+
+        config.clearLogRecords();
+        config.modifyResourceFile("db/migration/V1.0.0__Quarkus.sql", s -> s + "\nalter table FRUIT add column color VARCHAR;");
+        config.modifyResourceFile("application.properties", s -> s + "\nquarkus.flyway.validate-on-migrate=true");
+
+        RestAssured.get("/flyway/current-version").thenReturn();
```

Comment:
What do you with the value returned here?

Reply:
I should look at that. I assumed I had to call some endpoint in order to trigger the reload of the app. But maybe that is wrong or there is a better way to achieve that?

Reply:
I've now replaced this with a `RestAssured.get("/")` call to trigger the application restart (I am not aware of any better way to do this) and instead verify the response to `GET /flyway/current-version` inside the Awaitility assertion.